oak/server/rust/oak_runtime/src/runtime/mod.rs

```diff
@@ -596,19 +596,17 @@ impl Runtime {
     }
 
     // TODO(#970): Improve the documentation.
-    /// Waits on a slice of `ChannelHalf`s, blocking until one of the following conditions:
+    /// Waits on a slice of `ChannelHalf`s, until at least one of the channels is able to read a
+    /// message, provided that all of the channels are valid; otherwise returns immediately.
```

Comment:
IIUC it will stop waiting if a channel becomes orphan, which means that "waits until at least one of the channels is able to read a message" is not always true.